server/src/server.js

```diff
@@ -221,11 +221,19 @@ app.use(function(req, res, next) {
     }
     authInfo.abTests = abTests;
   }
+
   if (authInfo.deviceId) {
+    if (!isValidDeviceId(authInfo.deviceId)) {
+      next(new Error("Bad deviceId in login"));
```

Comment:
This doesn't actually abort the request. I think just `throw new Error(...)` would work fine here? Otherwise this should be followed by a return.

Reply:
Updated by following with a return